modules/tinymce/src/themes/silver/test/ts/browser/editor/scrolling/EditorInScrollingContainerTest.ts

```diff
@@ -83,542 +99,755 @@ describe('browser.tinymce.themes.silver.editor.scrolling.EditorInScrollingContai
+      Assertions.assertStructure(
+        'Element should undock',
+        ApproxStructure.build((s, str, _arr) => s.element('div', {
+          styles: {
+            position: str.none()
+          }
+        })),
+        header
+      );
+    }
+  );
+
+  const pWaitUntilUndockedAbsolute = (header: SugarElement<HTMLElement>): Promise<void> => Waiter.pTryUntil(
+    'Waiting for sticky element to undock',
+    () => {
+      Assertions.assertStructure(
+        'Element should undock',
+        ApproxStructure.build((s, str, _arr) => s.element('div', {
+          styles: {
+            position: str.is('absolute')
+          }
+        })),
+        header
+      );
+    }
+  );
+
+  const pWaitUntilDockedAtTop = (header: SugarElement<HTMLElement>, optTop: Optional<number>): Promise<void> => Waiter.pTryUntil(
+    'Waiting for sticky element to dock',
+    () => {
+      Assertions.assertStructure(
+        'Element should dock to the top',
+        ApproxStructure.build((s, str, _arr) => s.element('div', {
+          styles: {
+            position: str.is('fixed'),
+            ...(optTop.map((top) => ({
+              // Allow 5px of error.
+              top: str.measurement(top, 'px', 5)
+            })).getOr({ }))
+          }
+        })),
+        header
+      );
+    }
+  );
+
+  const pWaitUntilAppears = (header: SugarElement<HTMLElement>): Promise<void> => Waiter.pTryUntil(
+    'Waiting for sticky toolbar to appear',
+    () => {
+      Assertions.assertStructure(
+        'Sticky toolbar should appear',
+        ApproxStructure.build((s, str, arr) => s.element('div', {
+          classes: [
+            // The first time it appears, it doesn't get fade-in, so we just look for non-fadeout
+            // for now.
+            arr.not(ui.editor.stickyHeaderInvisible.className),
+            // TINY-9408: In classic mode, the transition class isn't going away because
+            // there is another transition style clobbering it. Reinstate the `arr.not` check
+            // when the bug is fixed.
+            // arr.not(ui.editor.stickyHeaderTransitioning.className)
+          ]
+        })),
+        header
+      );
+    }
+  );
+
+  const pWaitUntilDisappears = (header: SugarElement<HTMLElement>): Promise<void> => Waiter.pTryUntil(
+    'Waiting for sticky toolbar to disappear',
+    () => {
+      Assertions.assertStructure(
+        'Sticky toolbar should disappear',
+        ApproxStructure.build((s, str, arr) => s.element('div', {
+          classes: [
+            arr.has(ui.editor.stickyHeaderInvisible.className),
+            // TINY-9408: In classic mode, the transition class isn't going away because
+            // there is another transition style clobbering it. Reinstate the `arr.not` check
+            // when the bug is fixed.
+            // arr.not(ui.editor.stickyHeaderTransitioning.className)
+          ]
+        })),
+        header
+      );
+    }
+  );
+
+  const pWaitUntilTriggersEvent = (
```

Comment:
Yeah, this should be a generic function also there are others that collect events over time etc so could be moved to say a mcagar EditorEvents module or something but maybe out of scope for this PR.

Reply:
I have logged a new ticket to move this to a generic function in McAgar. TINY-9559